### PR TITLE
Replace customerrors with go-multierror

### LINKED ISSUE #751: Replace multierror implementation with go-multierror
Terratest currently implements its own [multierror package](https://github.com/gruntwork-io/terratest/blob/master/modules/customerrors/multierror.go). We should scrap this and use [go-multierror](https://github.com/hashicorp/go-multierror) instead.

### PR DESCRIPTION
**Pull request by urohit011**
This pull request resolves #751 by replacing the use of [customerrors](https://github.com/gruntwork-io/terratest/tree/master/modules/customerrors) with [go-multierror](https://github.com/hashicorp/go-multierror)

Link to test result gist -> [gist](https://gist.github.com/urohit011/54e35984779139de5fc54bd329c12892)

The test github.com/gruntwork-io/terratest/test/terraform_scp_example_test.go fails, but it does not look like it's due to the change in this commit.

### CODE REVIEW

File: modules/aws/ec2-files.go
Comment (infraredgirl) [inline modules/aws/ec2-files.go:215]:
```suggestion
	var errorsOccurred = new(multierror.Error)
```

File: modules/aws/ec2-files.go
Comment (infraredgirl) [inline modules/aws/ec2-files.go:234]:
```suggestion
	return errorsOccurred.ErrorOrNil()
```

File: modules/packer/packer.go
Comment (infraredgirl) [inline modules/packer/packer.go:58]:
```suggestion
	var errorsOccurred = new(multierror.Error)
```

File: modules/packer/packer.go
Comment (infraredgirl) [inline modules/packer/packer.go:79]:
```suggestion
	return artifactNameToArtifactId, errorsOccurred.ErrorOrNil()
```

File: modules/ssh/ssh.go
Comment (infraredgirl) [inline modules/ssh/ssh.go:167]:
```suggestion
	var errorsOccurred = new(multierror.Error)
```

File: modules/ssh/ssh.go
Comment (infraredgirl) [inline modules/ssh/ssh.go:185]:
```suggestion
	return errorsOccurred.ErrorOrNil()
```

**Review by infraredgirl — COMMENTED** [review]
Thanks for the contribution, the code looks good!

I left a couple of suggestions inline:
- Using `new` is more readable when creating a new multierror
- We should always return `.ErrorOrNil()` as that takes care to return `nil` in the case there were no errors

**Comment by urohit011** [comment]
Thanks for the suggestions @infraredgirl, I will keep them in mind.

**Comment by infraredgirl** [comment]
Thanks for the updates, Rohit! The code LGTM now, but tests seem to be failing. This could be due to some unrelated test instability we have hit lately. There's a [PR](https://github.com/gruntwork-io/terratest/pull/765) open that should stabilize tests a bit, I'll wait for that to merge before merging this. Thanks for your patience!

**Comment by urohit011** [comment]
Thanks for the update, Ana. Sorry I saw your msg late.

**Comment by robmorgan** [comment]
FYI - #765 is now merged.

**Comment by infraredgirl** [comment]
Thanks Rob! I re-ran the tests and the only failure is in helm tests which were not affected by this change.

I'll go ahead and merge - thanks again for the contribution @urohit011 !

**Review by infraredgirl — APPROVED** [review]

**Comment by urohit011** [comment]
Thank you Ana and Rob :)

**Pull request merged**
